### PR TITLE
Fixes after tests

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- 
- 
- 
  
## Proposed changes
- 
- 
- 

## Testing procedure
- 
- 
- 

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/headerCheck/headerSignatureVerify.go [hunk 1]
```diff
@@ -171,7 +178,16 @@ func (hsv *HeaderSigVerifier) VerifySignature(header data.HeaderHandler) error {
 		return err
 	}
 
-	return hsv.VerifySignatureForHash(header, hash, header.GetPubKeysBitmap(), header.GetSignature())
+	bitmap := header.GetPubKeysBitmap()
+	sig := header.GetSignature()
+	if hsv.enableEpochsHandler.IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, headerCopy.GetEpoch()) {
```
Comment (ssd04) [inline process/headerCheck/headerSignatureVerify.go:183]: i think that header signature verifier component should not know if we want to handle current or previous header info; can we use only `VerifySingatureForHash` and decide from the caller which header info to use?

Comment (ssd04) [reply process/headerCheck/headerSignatureVerify.go:183]: how about other components where `VerifySignature` is being used? does this also fit with the check from process interceptors?

File: process/headerCheck/headerSignatureVerify.go [hunk 2]
```diff
@@ -171,7 +178,16 @@ func (hsv *HeaderSigVerifier) VerifySignature(header data.HeaderHandler) error {
...
+		headerCopy, hash, sig, bitmap, err = hsv.getPrevHeaderInfo(headerCopy)
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply process/headerCheck/headerSignatureVerify.go:184]: I will start working on the change regarding option b) so we can simplify here and for cross chain proofs as well.

File: process/headerCheck/headerSignatureVerify.go (hunk 1 above)
Comment (sstanculeanu) [reply process/headerCheck/headerSignatureVerify.go:183]: yes, it should be compatible in all places. We can keep it as it is for compatibility. After this release, a refactor will come in the next one probably and completely remove EquivalentMessagesFlag and only keep the new version.

File: consensus/spos/bls/subroundSignature.go [hunk 3]
```diff
@@ -299,7 +299,7 @@ func (sr *subroundSignature) doSignatureConsensusCheck() bool {
 	}
 	isJobDoneByConsensusNode := !isSelfLeader && isSelfInConsensusGroup && selfJobDone && multiKeyJobDone
 
-	isSubroundFinished := !isSelfInConsensusGroup || isJobDoneByConsensusNode || isJobDoneByLeader
+	isSubroundFinished := isJobDoneByConsensusNode || isJobDoneByLeader
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply consensus/spos/bls/subroundSignature.go:302]: this is already done on L261 as an early exit. It was part of the subroundSignature changes PR: https://github.com/multiversx/mx-chain-go/pull/5760/files#diff-4f4adc2b25d7edb3422708759a12cbcc7be62a34a1c0970a8339a4a531ddf385R261

**Review by AdoAdoAdo — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**